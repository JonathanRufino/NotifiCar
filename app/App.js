import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Platform, AsyncStorage, AppState, Alert } from 'react-native';
import FCM, {
    FCMEvent, 
    RemoteNotificationResult,
    WillPresentNotificationResult, 
    NotificationType,
} from 'react-native-fcm';
import { Actions } from 'react-native-router-flux';

import Routes from './routes/index';
import reduxStore from './redux/store';

registerKilledListener();

class App extends Component {
    componentDidMount() {
        FCM.requestPermissions()
            .then(() => {
                console.log('notification permission garanted');
                registerAppListener();
            })
            .catch(() => console.log('notification permission rejected'));
    }

    render() {
        return (
            <Provider store={reduxStore}>
                <Routes /> 
            </Provider>
        );
    }
}

export default App;

// these callback will be triggered even when app is killed
function registerKilledListener() {
    FCM.on(FCMEvent.Notification, notif => {
        AsyncStorage.setItem('lastNotification', JSON.stringify(notif));

        FCM.presentLocalNotification({
            title: Platform.OS === 'ios' ? notif.aps.alert.title : notif.fcm.title,
            body: Platform.OS === 'ios' ? notif.aps.alert.body : notif.fcm.body,
            sound: 'default',
            priority: 'high',
            vibrate: 300,
            wake_screen: true,
            show_in_foreground: true,
            click_action: 'com.jldevs.notificar', // for ios
        });

        if (notif.opened_from_tray) {
            setTimeout(() => {
                if (notif._actionIdentifier === 'reply') {
                    if (AppState.currentState !== 'background') {
                        // Alert.alert(`User replied ${JSON.stringify(notif._userText)}`);
                        Alert.alert('User replied');
                    } else {
                        AsyncStorage.setItem('lastMessage', JSON.stringify(notif._userText));
                    }
                }
                if (notif._actionIdentifier === 'view') {
                    Alert.alert('User clicked View in App');
                }
                if (notif._actionIdentifier === 'dismiss') {
                    Alert.alert('User clicked Dismiss');
                }
            }, 1000);
        }
    });
}

// these callback will be triggered only when app is foreground or background
function registerAppListener() {
    FCM.on(FCMEvent.Notification, notif => {
        FCM.presentLocalNotification({
            title: Platform.OS === 'ios' ? notif.aps.alert.title : notif.fcm.title,
            body: Platform.OS === 'ios' ? notif.aps.alert.body : notif.fcm.body,
            sound: 'default',
            priority: 'high',
            vibrate: 300,
            wake_screen: true,
            show_in_foreground: true,
            click_action: 'com.jldevs.notificar', // for ios
        });

        if (Platform.OS === 'ios'
                && notif._notificationType === NotificationType.WillPresent
                && !notif.local_notification) {
            // this notification is only to decide if you want to 
            // show the notification when user if in foreground.
            // usually you can ignore it. just decide to show or not.
            notif.finish(WillPresentNotificationResult.All);
            return;
        }
  
        if (notif.opened_from_tray) {
            if (notif.targetScreen === 'detail') {
                setTimeout(() => {
                    Actions.login();
                }, 500);
            }
            setTimeout(() => {
                // Alert.alert(`User tapped notification\n${JSON.stringify(notif)}`);
                Actions.login();
            }, 500);
        }
  
        if (Platform.OS === 'ios') {
            //optional
            //iOS requires developers
            //to call completionHandler to end notification process. If you do not call it your background remote notifications could be throttled, to read more about it see the above documentation link.
            //This library handles 
            // it for you automatically with default behavior (for remote notification, finish with NoData; for WillPresent, finish depend on "show_in_foreground"). However if you want to return different result, follow the following code to override
            //notif._notificationType is available for iOS platfrom
            switch (notif._notificationType) {
                case NotificationType.Remote:
                    notif.finish(RemoteNotificationResult.NewData); //other types available: RemoteNotificationResult.NewData, RemoteNotificationResult.ResultFailed
                    break;
                case NotificationType.NotificationResponse:
                    notif.finish();
                    break;
                case NotificationType.WillPresent:
                    notif.finish(WillPresentNotificationResult.All); //other types available: WillPresentNotificationResult.None
                    // this type of notificaiton will be called only when you are in foreground.
                    // if it is a remote notification, don't do any app logic here. Another notification callback will be triggered with type NotificationType.Remote
                    break;
                default:
                    break;
            }
        }
    });
}
