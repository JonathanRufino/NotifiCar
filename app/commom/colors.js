const Colors = {
    WHITE: '#FFF',
    BLACK: '#000',
    BLUE: '#1E88E5',
    RED: '#FF0000',
    GRAY: '#aeb2b7',
    LIGHT_GRAY: '#d6d8db',
    DARK_GRAY: '#686868',
    TRANSPARENT: 'transparent',
    GREEN: '#115E54',
    GREEN_DARK: '#114D44',
    RED_DARK: '#D23438',
    RED_MIDDLE: '#Ed3237',
    RED_LIGHT: '#Eb7575',
    FACEBOOK_BLUE: '#3B5998',
};

export { Colors };
