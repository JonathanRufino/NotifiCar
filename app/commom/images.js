const Images = {
    LOGIN_BACKGROUND: require('../assets/images/login_background.png'),
    LOGO_WHITE: require('../assets/images/logo-white.png'),
    LOGO_SPLASHSCREEN: require('../assets/images/logo-splashscreen.png'),
    ICON_ALARM: require('../assets/images/alarm.png'),
    ICON_LIGHTS: require('../assets/images/lights.png'),
    ICON_WARNING: require('../assets/images/warning.png'),
    ICON_WINDOW: require('../assets/images/window.png'),
    ICON_CAR: require('../assets/images/car.png'),
    ICON_ATTENTION: require('../assets/images/attention.png'),
    ICON_DOOR_OPEN: require('../assets/images/door_open.png'),
    ICON_CAR_BLOCKING: require('../assets/images/car_blocking.png'),
    ICON_FLAT_TIRE: require('../assets/images/flat_tire.png'),
    ICON_TRASH: require('../assets/images/icon-trash.png'),
};

export { Images };
