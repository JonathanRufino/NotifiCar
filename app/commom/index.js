export { Colors } from './colors';
export { Texts } from './texts';
export { Regexes } from './regexes';
