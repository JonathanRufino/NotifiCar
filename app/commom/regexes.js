const Regexes = {
    LICENSE_PLATE: /^[a-zA-Z]{3}-\d{4}$/,
};

export { Regexes };    
