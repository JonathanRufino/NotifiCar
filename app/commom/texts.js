const Texts = {
    Titles: {
        REGISTER_VEHICLE: 'Cadastrar Veículo',
    },
    Errors: {
        EMPTY_LICENSE_PLATE: 'Informe a placa do veículo',
        INVALID_LICENSE_PLATE: 'Informe uma placa válida',
    },
    Placeholders: {
        LICENSE_PLATE: 'AAA-0000',
    },
    Buttons: {
        REGISTER: 'Cadastrar',
    }
};

export { Texts };
