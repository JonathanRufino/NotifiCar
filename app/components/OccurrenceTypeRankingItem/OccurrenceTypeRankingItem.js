import React from 'react';
import { View, Text, Image } from 'react-native';

import styles from './styles';

const OccurrenceTypeRanking = ({ occurrenceType, image }) => (
    <View style={styles.container}>
        <Image style={styles.icon} source={image} />

        <View style={styles.typeContainer}>
            <Text style={styles.type}>
                { occurrenceType.type }
            </Text>
        </View>
        <View style={styles.countContainer}>
            <Text style={styles.count}>
                { occurrenceType.occurrences_count }
            </Text>
        </View>
    </View>
);

export default OccurrenceTypeRanking;
