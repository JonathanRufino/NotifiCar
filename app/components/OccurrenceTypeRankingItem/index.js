import OccurrenceTypeRankingItem from './OccurrenceTypeRankingItem';

export default OccurrenceTypeRankingItem;
