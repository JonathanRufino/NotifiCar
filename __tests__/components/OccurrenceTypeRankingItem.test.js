import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import OccurrenceTypeRankingItem from '../../app/components/OccurrenceTypeRankingItem';
import { Images } from '../../app/commom';

describe('<OccurrenceTypeRankingItem />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <OccurrenceTypeRankingItem
                occurrenceType='Pneu Furado'
                image={Images.ICON_FLAT_TIRE}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
