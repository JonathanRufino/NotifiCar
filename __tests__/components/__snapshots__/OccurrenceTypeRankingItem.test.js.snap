// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OccurrenceTypeRankingItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 20,
      "marginVertical": 10,
      "width": 750,
    }
  }
>
  <Image
    source={Object {}}
    style={
      Object {
        "height": 32,
        "width": 32,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 7,
        "justifyContent": "center",
        "marginLeft": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 22,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 2,
        "justifyContent": "space-around",
        "marginLeft": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#63C851",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
